src/lib/io/include/sol-iio.h

```diff
@@ -77,6 +77,7 @@ typedef struct sol_iio_config {
     const void *data; /**< User defined data to be sent to sol_iio_reader_cb */
     int buffer_size; /**< The size of reading buffer. 0: use device default; -1: disable buffer and readings will be performed on channel files on sysfs. */
     int sampling_frequency; /**< Device sampling frequency. -1 uses device default */
+    char sampling_frequency_name[PATH_MAX]; /**< Sampling frequency sysfs node name. Some drivers expose the sampling frequency that is shared by channel type. Such as in_magn_sampling_frequency, in_accel_sampling_frequency. */
```

Comment:
`PATH_MAX` appears overkill. Knowing that this is only part of name, why not `NAME_MAX`?